### PR TITLE
Implement desktop dropdown in meganav

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Implements the HTML and JS needed for dropdown functionality in the meganav. Based on the work done by Bartek in the vanilla team here: https://github.com/canonical/vanilla-framework/pull/5213
- Adds the navigation to build
- Some basic refactoring to get the JS to work in this instance and modulising to make it easier to understand 

## QA

- You should follow these QA steps on the [regular navigation](https://canonical-com-1301.demos.haus/) and [reduced navigation](https://canonical-com-1301.demos.haus/data). Mobile can be ignored for this PR.
- Click on a dropdown button (ie. products, solutions, etc) and see that the corresponding dropdown opens, *note* the content is a place holder. Click again to close.
- Click on a dropdown button, click outside the navigation and see the dropdown closes
- Click on a dropdown button, open the search and see the dropdown closes
- When a dropdown opens, it should side down appearing from underneath the top navigation.
- When a dropdown closes, it should slide down disappearing underneath the top navigation.
- When switching between different dropdowns (ie. products, solutions, etc), no animation should be played
- The dropdown should match the one in the [vanilla example](https://vanilla-framework-5213.demos.haus/docs/examples/patterns/navigation/dropdown-full-width?theme=light)

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-12080

### CODE REVIEW

File: static/js/navigation/search.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+export function toggleSearch() {
+  const isOpen = navigation.classList.contains("has-search-open");
+
+  if (!isOpen) {
+    closeAllNavigationItems();
+    openSearch();
+  } else {
+    closeAllNavigationItems();
+  }
```
Comment (britneywwc) [inline static/js/navigation/search.js:16]: The conditional expression can be refactored 
```suggestion
  closeAllNavigationItems();
  if (!isOpen) {
    openSearch();
  }
```

File: static/js/navigation/main.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      target.setAttribute("aria-hidden", "true");
+      toggle.parentNode.classList.remove("is-active");
+      toggle.parentNode.parentNode.classList.remove("is-active");
+      navigation.classList.remove("has-menu-open");
+    }, 100);
+  }
+}
+
+/**
+ * Sets tabindex for appropriate navigation items to allow keybaord navigation
```
Comment (britneywwc) [inline static/js/navigation/main.js:117]:
```suggestion
 * Sets tabindex for appropriate navigation items to allow keyboard navigation
```

**Review by britneywwc — APPROVED** [review]
Looks good +1 🔥 The secondary-navigation.js seems sound to me but I can't QA that just yet since it's only on mobile view. I'll just give it a +1 until that can be tested further

File: static/js/navigation/search.js (hunk 1 above)
Comment (petesfrench) [reply static/js/navigation/search.js:16]: nice :fire:

**Comment by petesfrench** [comment]
@britneywwc Thanks for the review. Yes, you can ignore most mobile changes. They will be addressed in later PRs.

For the secondary-navigation.js, you can test it by visiting https://canonical-com-1301.demos.haus/data on mobile, you should be able to open/close the dropdown:
image

**Comment by britneywwc** [comment]
Cool, just tested out the secondary nav and it looks great

**Pull request merged**